components/rating/rating.tsx

```diff
@@ -0,0 +1,124 @@
+import { Icon, Props as IconProps } from '@geist-ui/react-icons'
+import React, { useEffect, useMemo, useState } from 'react'
+
+import { GeistUIThemes } from '../themes/presets'
+import { NormalTypes } from '../utils/prop-types'
+import Star from '@geist-ui/react-icons/star'
+import useTheme from '../use-theme'
+import withDefaults from '../utils/with-defaults'
+
+interface Props {
+  type?: NormalTypes
+  className?: string
+  icon?: (props: IconProps) => React.ReactElement<Icon>
```

Comment:
Is there any way we can change this to a normal `JSX.Element`?
Remove the function, reducing complexity and removing the type dependency on `Icon / IconProps`.

Reply:
For using `JSX.Element` as a props, you can refer to the type design of the `Button` component.

Reply:
```suggestion
  icon?: JSX.Element
```
I think straight up allowing `JSX.Element` here is fine. This would remove the `Icon / IconProps` dependency completely and open up the interface for customized elements. Do you agree?

I will create a complete re-request for a review, so there's no need to incorporate any suggestions directly.